mongo/mongo.go

```diff
@@ -267,39 +196,7 @@ func ensureNoDollarKey(doc bsoncore.Document) error {
 	return nil
 }
 
-func transformAggregatePipeline(registry *bsoncodec.Registry, pipeline interface{}) (bsonx.Arr, error) {
-	pipelineArr := bsonx.Arr{}
-	switch t := pipeline.(type) {
-	case bsoncodec.ValueMarshaler:
-		btype, val, err := t.MarshalBSONValue()
-		if err != nil {
-			return nil, err
-		}
-		if btype != bsontype.Array {
-			return nil, fmt.Errorf("ValueMarshaler returned a %v, but was expecting %v", btype, bsontype.Array)
-		}
-		err = pipelineArr.UnmarshalBSONValue(btype, val)
-		if err != nil {
-			return nil, err
-		}
-	default:
-		val := reflect.ValueOf(t)
-		if !val.IsValid() || (val.Kind() != reflect.Slice && val.Kind() != reflect.Array) {
-			return nil, fmt.Errorf("can only transform slices and arrays into aggregation pipelines, but got %v", val.Kind())
-		}
-		for idx := 0; idx < val.Len(); idx++ {
-			elem, err := transformDocument(registry, val.Index(idx).Interface())
-			if err != nil {
-				return nil, err
-			}
-			pipelineArr = append(pipelineArr, bsonx.Document(elem))
-		}
-	}
-
-	return pipelineArr, nil
-}
-
-func transformAggregatePipelinev2(registry *bsoncodec.Registry, pipeline interface{}) (bsoncore.Document, bool, error) {
+func transformAggregatePipeline(registry *bsoncodec.Registry, pipeline interface{}) (bsoncore.Document, bool, error) {
```

Comment:
I thought briefly about making the return type `(bsoncore.Array, bool, error)` to use the new type, but it does require a larger refactor.

Reply:
I am not sure if that adds value. But if this works as-is, that seems fine to me.